pylint/checkers/typecheck.py

```diff
@@ -1161,18 +1161,23 @@ def _check_assignment_from_function_call(self, node):
 
         # Make sure that it's a valid function that we can analyze.
         # Ordered from less expensive to more expensive checks.
-        # pylint: disable=too-many-boolean-expressions
         if (
-            not function_node.is_function
+            function_node.decorators is not None
+            or self._is_ignored_function(function_node)
             or isinstance(function_node, nodes.AsyncFunctionDef)
-            or function_node.decorators
-            or function_node.is_generator()
-            or function_node.is_abstract(pass_is_abstract=False)
             or utils.is_error(function_node)
-            or not function_node.root().fully_defined()
         ):
             return
 
+        if self._list_sort_method(
+            node.value
+        ):  # false-negative case fix, see issue #5722
+            self.add_message("assignment-from-none", node=node)
+            return
+
+        if not function_node.root().fully_defined():
+            return
+
         returns = list(
             function_node.nodes_of_class(nodes.Return, skip_klass=nodes.FunctionDef)
         )
```

Comment:
I'm wondering why we could not make the generic part also work for list.sort() though, what went wrong ?

Reply:
I know why.. I debugged it.
It fails because `function_node.root().fully_defined()` returns `False`.
It returns `False` because `astroid` searches for a return value in the **abstract** definition of the `sort` function, in `builtins.py`:

```
    def sort(self, *args, **kwargs): # real signature unknown
        """
        Sort the list in ascending order and return None.
        
        The sort is in-place (i.e. the list itself is modified) and stable (i.e. the
        order of two equal elements is maintained).
        
        If a key function is given, apply it once to each list item and sort them,
        ascending or descending, according to their function values.
        
        The reverse flag can be set to sort in descending order.
        """
        pass
```

I don't think it's possible to retrieve the `astroid.FunctionDef` of the correct function..
If it is possible then we should fix the redirection itself.. 
It might raise some false-positives though,..

Anyone knows if it's possible to do so in `astroid`?

Reply:
seems from [this SO answer ](https://stackoverflow.com/questions/52081316/how-to-view-the-implementation-of-pythons-built-in-functions-in-pycharm) that there is there is no actual python implementation, that the C implementation already built-in inside the interpreter...

Reply:
Thanks @orSolocate for this thorough search!